server/aws-lsp-identity/src/sso/cache/fileSystemSsoCache.ts

```diff
@@ -0,0 +1,107 @@
+        })
+    }
+
+    async setSsoClientRegistration(
+        clientName: string,
+        ssoRegion: string,
+        ssoStartUrl: string,
+        clientRegistration: SsoClientRegistration
+    ): Promise<void> {
+        if (!clientName || !ssoRegion || !ssoStartUrl || !ssoClientRegistrationDuckTyper.eval(clientRegistration)) {
+            return
+        }
+
+        const id = toSsoClientRegistrationCacheId(clientName, ssoRegion, ssoStartUrl)
+        await tryAsync(
+            () => writeSsoObjectToFile(id, clientRegistration),
+            error => AwsError.wrap(error, AwsErrorCodes.E_CANNOT_WRITE_SSO_CACHE)
+        )
+    }
+
+    async getSsoToken(ssoSessionName: string): Promise<SSOToken | undefined> {
+        return await tryGetAsync(async () => {
+            const ssoToken = await getSSOTokenFromFile(ssoSessionName)
+            return ssoTokenDuckTyper.eval(ssoToken) ? ssoToken : undefined
+        })
+    }
+
+    async setSsoToken(ssoSessionName: string, ssoToken: SSOToken): Promise<void> {
+        if (!ssoSessionName || !ssoTokenDuckTyper.eval(ssoToken)) {
+            return
+        }
+
+        tryAsync(
+            () => writeSsoObjectToFile(ssoSessionName, ssoToken),
+            error => AwsError.wrap(error, AwsErrorCodes.E_CANNOT_WRITE_SSO_CACHE)
+        )
+    }
+}
+
+async function tryGetAsync<R>(tryIt: () => Promise<R | undefined>): Promise<R | undefined> {
+    try {
+        return await tryIt()
+    } catch (error) {
+        // Error codes are consistent across OSes (Windows is converted to libuv error codes)
+        // https://nodejs.org/api/errors.html#errorerrno
+        if ((error as SystemError)?.code === 'ENOENT') {
+            return Promise.resolve(undefined)
+        }
+
+        throw AwsError.wrap(error as Error, AwsErrorCodes.E_CANNOT_READ_SSO_CACHE)
+    }
+}
+
+function toSsoClientRegistrationCacheId(clientName: string, ssoRegion: string, ssoStartUrl: string): string {
+    return JSON.stringify({
+        region: ssoRegion,
+        startUrl: ssoStartUrl,
+        tool: clientName,
+    })
+}
+
+// Based on:
+// https://github.com/smithy-lang/smithy-typescript/blob/main/packages/shared-ini-file-loader/src/getSSOTokenFromFile.ts
+async function getSsoClientRegistrationFromFile(id: string): Promise<SsoClientRegistration> {
+    const filepath = getSSOTokenFilepath(id)
+    const json = await readFile(filepath, 'utf8')
+    return JSON.parse(json) as SsoClientRegistration
+}
+
+// Based on:
+// https://github.com/aws/aws-sdk-js-v3/blob/6e61f0e78ff7a9e3b1f2cd651bde5fc656d85ba9/packages/token-providers/src/writeSSOTokenToFile.ts
+async function writeSsoObjectToFile(id: string, ssoObject: SSOToken | SsoClientRegistration): Promise<void> {
+    try {
+        const filepath = getSSOTokenFilepath(id)
+        const json = JSON.stringify(ssoObject, null, 2)
+        return await writeFile(filepath, json)
+    } catch {
+        // Writing to cache is best effort, based on:
```

Comment:
Do we at least want to log something?

Reply:
Yes, that is addressed at the bottom of the PR comments above "Logging throughout the identity server will be added in a future PR (IDE-14689). This will notify users of cache issues without interrupting their work." 

There is a task(s) to add logging (w/o PII) throughout the server, but to support the deadlines in the related project, I have to get GetSsoToken out first.  

There will additionally be an update to the cache to remove token files for the InvalidateSsoToken, but that is coming as planned/scheduled near future work as well.

Reply:
IMO, adding logging right away should not be a big effort. Troubleshooting server without logs can take more time than adding logging in process of implementation at the end.